From: John Ogness <john.ogness@linutronix.de>
Date: Sun, 21 Jul 2024 19:36:53 +0000
Subject: [PATCH 51/54] printk: nbcon: Assign nice -20 for printing threads

It is important that console printing threads are scheduled
shortly after a printk call and with generous runtime budgets.

Signed-off-by: John Ogness <john.ogness@linutronix.de>
Reviewed-by: Petr Mladek <pmladek@suse.com>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 kernel/printk/nbcon.c  |    6 ++++++
 kernel/printk/printk.c |    6 ++++++
 2 files changed, 12 insertions(+)

--- a/kernel/printk/nbcon.c
+++ b/kernel/printk/nbcon.c
@@ -1312,6 +1312,12 @@ bool nbcon_kthread_create(struct console
 
 	con->kthread = kt;
 
+	/*
+	 * It is important that console printing threads are scheduled
+	 * shortly after a printk call and with generous runtime budgets.
+	 */
+	sched_set_normal(con->kthread, -20);
+
 	return true;
 }
 
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -3554,6 +3554,12 @@ static bool legacy_kthread_create(void)
 
 	printk_legacy_kthread = kt;
 
+	/*
+	 * It is important that console printing threads are scheduled
+	 * shortly after a printk call and with generous runtime budgets.
+	 */
+	sched_set_normal(printk_legacy_kthread, -20);
+
 	return true;
 }
 
